sources/osgShader/ShaderGenerator.js

```diff
@@ -295,6 +366,15 @@ ShaderGenerator.prototype = {
             program.generated = true;
 
             this._cache[hash] = program;
+
+            // first async compilation
+            program.apply(state, !synchronizedCompilation);
```

Comment:
not sure apply need an extra argument if program has extra data like the _placeHolder or asyncCompilation flag